src/charm_state.py

```diff
@@ -165,6 +162,54 @@ def _valid_storage_size_str(size: str) -> bool:
     return size[-3:] in valid_suffixes and size[:-3].isdigit()
 
 
+OPENSTACK_CLOUDS_YAML_CONFIG_NAME = "experimental-openstack-clouds-yaml"
+CLOUDS_YAML_PATH = Path(Path.home() / ".config/openstack/clouds.yaml")
+
+
+def _validate_openstack_cloud_config(cloud_config: dict) -> None:
+    """Validate the format of the cloud configuration.
+
+    Args:
+        cloud_config: The configuration in clouds.yaml format to validate.
+
+    Raises:
+        InvalidConfigError: if the format of the config is invalid.
+    """
+    # dict of format: {clouds: <cloud-name>: <cloud-config>}
+    try:
+        clouds = list(cloud_config["clouds"].keys())
+    except KeyError as exc:
+        raise OpenStackInvalidConfigError("Invalid clouds.yaml.") from exc
```

Comment:
Perhaps it could specifically say that the `clouds` key is missing?

Reply:
That is indeed much more helpful. Thank you!